Add extensible benchmarking system with `is_prime` benchmark

partylikeits1983 opened the pull request:
This PR adds a `benches/` directory with an extensible framework for benchmarking Miden compiler programs. The system automatically compiles Rust programs with `cargo miden build`, executes them with `midenc run`, and measures VM cycles and performance metrics.

**Usage:**
```bash
# Run is_prime benchmark
cargo run -p miden-compiler-benches --bin is_prime

# Test different inputs and iterations
cargo run -p miden-compiler-benches --bin is_prime -- --input 97 --iterations 3
```

The framework is designed to be easily extended for benchmarking other Miden programs by adding new binaries to `benches/src/`.

bitwalker commented on benches/src/is_prime.rs:
```diff
@@ -0,0 +1,101 @@
+use miden_compiler_benches::BenchmarkRunner;
+
+fn main() -> Result<()> {
+    let matches = Command::new("is_prime_benchmark")
```
I'd recommend defining an argument structure, like so:

```rust
use clap::Parser;

#[derive(Parser)]
struct Config {
    /// The number to test for primality
    #[arg(short = 'i', long, value_name = "NUMBER", default_value = "29")]
    input: usize,
    /// Path to the is_prime source file
    #[arg(short = 's', long, value_name = "PATH", default_value = "examples/is_prime/src/lib.rs")]
    source: PathBuf,
    /// Number of iterations to run
    #[arg(short = 'n', long, value_name = "COUNT", default_value = "1")]
    iterations: usize,
}
```

Then you can parse the arguments with just:

```rust
fn main() -> anyhow::Result<()> {
    let config = Config::parse()?;
    ...
}
```

bitwalker reviewed (changes requested):
Looks good! Thanks for putting this together!

I have a couple of tweaks I'd like to make before merging, but they are small things.

partylikeits1983 commented:
I addressed feedback by renaming the crate to "midenc-benchmark-runner," adding a bench task to `Makefile.toml`, and updating the argument parsing in `benches/src/is_prime.rs`.
